cookbooks/bcpc-hadoop/recipes/hive_table_stat.rb

```diff
@@ -0,0 +1,46 @@
+#
+# Cookbook Name : bcpc-hadoop
+# Recipe Name : hive_table_stat
+# Description : create database for hive table statistics 
+#
+
+#mysql_connection_info = {
+#  :host     => '127.0.0.1',
+#  :username => 'root',
+#  :password => get_config('mysql-root-password')
+#}
+
+#mysql_database node["bcpc"]["hadoop"]["hive"]["hive_table_stats_db"] do
+#  connection mysql_connection_info 
+#  action :create
+#end
+
+#mysql_database_user node["bcpc"]["hadoop"]["hive"]["hive_table_stats_db_user"] do
+#  connection mysql_connection_info 
+#  password get_config("mysql-hive-table-stats-password") 
+#  action :create
+#end
+
+
+
+ruby_block "hive_table_stats_db" do
+  cmd = "mysql -uroot -p#{get_config('mysql-root-password')} -e"
+  privs = "ALL" # todo node[:bcpc][:hadoop][:hive_db_privs].join(",")
+  block do
+    if not system " #{cmd} 'SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = #{node["bcpc"]["hadoop"]["hive"]["hive_table_stats_db"]}' | grep -q #{node["bcpc"]["hadoop"]["hive"]["hive_table_stats_db"]}" then
+      code = <<-EOF
+        CREATE DATABASE #{node["bcpc"]["hadoop"]["hive"]["hive_table_stats_db"]};
+        GRANT #{privs} ON #{node["bcpc"]["hadoop"]["hive"]["hive_table_stats_db"]}.* TO '#{node["bcpc"]["hadoop"]["hive"]["hive_table_stats_db_user"]}'@'%' IDENTIFIED BY '#{get_config('mysql-hive-table-stats-password')}';
+        FLUSH PRIVILEGES;
+        USE metastore;
+        SOURCE /usr/lib/hive/scripts/metastore/upgrade/mysql/hive-schema-0.12.0.mysql.sql;
```

Comment:
Checking a VM cluster built from master, couldn't find the file in this location instead it is in ``/usr/hdp/current/hive-metastore/scripts/metastore/upgrade/mysql`` . Do we want to use an attribute instead of hardcoding it since the sql file name changes with different version of ``mysql`` used. This will also help with changing the location if it changes in the future.